cores/HardwareSerial.cpp

```diff
@@ -1,234 +1,239 @@
-            XMC_UART_CH_EnableEvent(_XMC_UART_config->channel, XMC_UART_CH_EVENT_TRANSMIT_BUFFER);
-            XMC_UART_CH_Transmit(_XMC_UART_config->channel,
-                                 _tx_buffer->_aucBuffer[_tx_buffer->_iTail]);
-            _tx_buffer->_iTail++;
-            if (_tx_buffer->_iTail >= SERIAL_BUFFER_SIZE)
-                _tx_buffer->_iTail = _tx_buffer->_iTail -
-                                     SERIAL_BUFFER_SIZE; // If iTail is larger than Serial Buffer
-                                                         // Size calculate the correct index value
-        }
-
-        while (_tx_buffer->_iTail == nextWrite)
-            ; // Spin locks if we're about to overwrite the buffer. This continues once the data is
-              // sent
-
-        _tx_buffer->_aucBuffer[_tx_buffer->_iHead] = uc_data;
-        _tx_buffer->_iHead = nextWrite;
-    } else {
-        // Make sure TX interrupt is enabled
-        XMC_UART_CH_EnableEvent(_XMC_UART_config->channel, XMC_UART_CH_EVENT_TRANSMIT_BUFFER);
-        // Bypass buffering and send character directly
-        XMC_UART_CH_Transmit(_XMC_UART_config->channel, uc_data);
-    }
-    return 1;
-}
-
-void HardwareSerial::IrqHandler(void) {
-    uint32_t status = XMC_UART_CH_GetStatusFlag(_XMC_UART_config->channel);
-
-    // Did we receive data?
-    if ((status & (XMC_UART_CH_STATUS_FLAG_ALTERNATIVE_RECEIVE_INDICATION |
-                   XMC_UART_CH_STATUS_FLAG_RECEIVE_INDICATION)) != 0U) {
-        XMC_UART_CH_ClearStatusFlag(_XMC_UART_config->channel,
-                                    (XMC_UART_CH_STATUS_FLAG_ALTERNATIVE_RECEIVE_INDICATION |
-                                     XMC_UART_CH_STATUS_FLAG_RECEIVE_INDICATION));
-
-        while (_XMC_UART_config->channel->RBUFSR &
-               (USIC_CH_RBUFSR_RDV0_Msk | USIC_CH_RBUFSR_RDV1_Msk))
-            _rx_buffer->store_char(XMC_UART_CH_GetReceivedData(_XMC_UART_config->channel));
-    }
-
-    // Do we need to keep sending data?
-    if ((status & XMC_UART_CH_STATUS_FLAG_TRANSMIT_BUFFER_INDICATION) != 0U) {
-        XMC_UART_CH_ClearStatusFlag(_XMC_UART_config->channel,
-                                    XMC_UART_CH_STATUS_FLAG_TRANSMIT_BUFFER_INDICATION);
-
-        if (_tx_buffer->_iTail != _tx_buffer->_iHead) {
-            XMC_UART_CH_Transmit(_XMC_UART_config->channel,
-                                 _tx_buffer->_aucBuffer[_tx_buffer->_iTail]);
-            _tx_buffer->_iTail++;
-            if (_tx_buffer->_iTail >= SERIAL_BUFFER_SIZE)
-                _tx_buffer->_iTail = 0;
-        } else {
-            // Mask off transmit interrupt so we don't get it any more
-            XMC_UART_CH_DisableEvent(_XMC_UART_config->channel, XMC_UART_CH_EVENT_TRANSMIT_BUFFER);
-        }
-    }
-}
-
-//****************************************************************************
-//                                 END OF FILE
-//****************************************************************************
+/*

```

Comment:
What has changed here?
Seems all deleted and added again. Not sure how the code is being handled git-wise. But we should try to make these diff show the actual changes, to allow us review effectively, and focus on the changes.
Also for the hil-unity-checks..yml file.

Reply:
I need to disable autocrlf for other project. I think problem is EOL inconsistency.
need  .gitattributes/ inputs from Architecture board 😉 
![image](https://github.com/user-attachments/assets/a8ff1eec-e35e-4604-b183-a51be420564c)

Reply:
fixed and added .gitattributes.
This timeout added for fix one of the i2c test.